packages/components/src/internal/components/domainproperties/list/ListDesignerPanels.tsx

```diff
@@ -106,22 +107,57 @@ class ListDesignerPanelsImpl extends React.PureComponent<Props & InjectedBaseDom
             });
     }
 
-    onFinish = () => {
-        const { setSubmitting } = this.props;
+    setAndValidateKeyDataAfterFieldImport = (): ListModel => {
         const { model } = this.state;
-        const isValid = model.isValid();
+        const fields = model.domain.fields;
+        const pkIndex = fields.findIndex(i => i.isPrimaryKey);
 
-        this.props.onFinish(isValid, this.saveDomain);
+        if (pkIndex > -1 && !model.keyType) {
+            const pkField = fields.get(pkIndex);
+            const keyName = pkField.get("name");
 
-        if (!isValid) {
-            const exception = !model.hasValidKeyType()
-                ? 'You must specify a key field for your list in the fields panel to continue.'
-                : model.domain.getFirstFieldError();
-            const updatedModel = model.set('exception', exception) as ListModel;
-            setSubmitting(false, () => {
-                this.setState(() => ({ model: updatedModel }));
-            });
+            const isAutoIncPk = PropDescType.isAutoIncrement(pkField.dataType);
+            const isInt = PropDescType.isInteger(pkField.dataType.rangeURI);
+            const isVarchar = PropDescType.isString(pkField.dataType.rangeURI);
+
+            if (isAutoIncPk) {
+                return(model.merge({ keyName, keyType: 'AutoIncrementInteger'}) as ListModel);
+            } else if (isInt) {
+                return(model.merge({ keyName, keyType: 'Integer'}) as ListModel);
+            } else if (isVarchar) {
+                return(model.merge({ keyName, keyType: 'Varchar'}) as ListModel);
+            } else {
+                return(model.merge({ keyType: 'invalid'}) as ListModel);
+            }
+        } else {
+            return(model);
         }
+    }
+
+    onFinish = () => {
+        const { setSubmitting } = this.props;
+        const model = this.setAndValidateKeyDataAfterFieldImport();
```

Comment:
@RosalineP I like this change, but I think we can call this setter method in the onDomainChange function so that it gets applied right after the file drop instead of waiting until the finish button is clicked. I have this change locally and will push shortly.